posts/2024-07-30-24.0.0.8-beta.adoc

```diff
@@ -0,0 +1,816 @@
+- link:https://github.com/jakartaee/nosql[Eclipse jNoSQL support]
+
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC>
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // //
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/28770
+// Contact/Reviewer: volosied,pnicolucci
+// // // // // // // //
+
+[#faces]
+== Bring your own Jakarta Faces implementation with Jakarta Faces Container 4.1
+
+Jakarta Faces is a Model-View-Controller (MVC) framework for building web applications. It offers many convenient features, such as state management and input validation.
+
+The Jakarta Faces Container 4.1 feature (`facesContainer-4.1`) enables you to bring your own Jakarta Faces API and implementation to Liberty. The alternative is to use the Liberty-provided `faces-4.1` feature, which provides the MyFaces 4.1 API and implementation.
+
+The Faces Container feature is updated in this beta release to support the 4.1 API and implementation JARs. If the 4.1 JARs were used with earlier features, errors occurred because the feature checks against the specification version listed within the jars.
+
+To use the `facesContainer-4.1` feature, add the following code to your `server.xml` file:
+
+[source,xml]
+----
+<featureManager>
+   <feature>facesContainer-4.1</feature>
+</featureManager>
+----
+
+For more information, see the following resources:
+
+* link:https://openliberty.io/docs/latest/reference/feature/facesContainer-4.0.html[Jakarta Faces Container 4.0] feature
+* link:https://openliberty.io/blog/2024/07/02/24.0.0.7-beta.html#faces41[Faces 4.1 beta blog announcement]
+* link:https://jakarta.ee/specifications/faces/4.1/[Faces 4.1 Javadocs and specification document].
+
+
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC>
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // //
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/28769
+// Contact/Reviewer: volosied,pnicolucci
+// // // // // // // //
+
+[#websocket]
+== Improve asynchronous handling with Jakarta WebSocket 2.2
+
+The Jakarta WebSocket 2.2 (`websocket-2.2`) feature enables communication for endpoints by using the Websocket protocol. It is an implementation of the Jakarta EE 11 Websocket 2.2 Specification.
+
+Although very few specification changes were listed for the 2.2 release, the main change from 2.1 is the introduction of the link:https://jakarta.ee/specifications/websocket/2.2/apidocs/server/jakarta/websocket/sendresult#getSession()[getSession()] method on the `SendResult` class. This method enables you to retrieve the original session that was used in the asynchronous remote call. This capability is useful for broadcasting messages to numerous listening clients.
+
+The next beta release of websocket-2.2 will limit the link:https://jakarta.ee/specifications/websocket/2.2/apidocs/server/jakarta/websocket/onmessage#maxMessageSize()[MaxMessageSize] to `Integer#MAX_VALUE` for the `OnMessage` annotation to avoid type errors.

```

Comment:
@volosied / @pnicolucci is this pre-announcing something in a future beta or is it included in this beta?  If in this beta, it should be updated not say `The next beta`. If it will be in the next beta, I think maybe this line should be removed?

Reply:
@jhanders34 this is a pre-announcement . If this is unusual to do, it could be removed and we could just do another websockets beta blog for 24009 beta

Reply:
The change is small which is way we decided to just say in the next beta.

Reply:
I've removed it, will announce in 24.0.0.9-beta

Reply:
It could be added as a known issue for WebSocket 2.2 in this beta release if something isn't the way it is supposed to be.  In other sections of this beta blog post you can see where that has been done.